torch/csrc/cuda/Module.cpp

```diff
@@ -519,32 +519,109 @@ PyObject* THCPModule_resetPeakMemoryStats(PyObject* _unused, PyObject* arg) {
   Py_RETURN_NONE;
 }
 
+struct Frame {
+  PyCodeObject* code;
+  int lasti;
+};
+
+struct StackContext : public c10::cuda::CUDACachingAllocator::Context {
+  std::vector<Frame> frames;
+  ~StackContext() {
+    for (auto& f : frames) {
+      Py_XDECREF((PyObject*)f.code);
+    }
+  }
+  static std::unique_ptr<c10::cuda::CUDACachingAllocator::Context> gather() {
+    py::gil_scoped_acquire acquire;
+    auto r = std::make_unique<StackContext>();
+    PyThreadState* tstate = PyThreadState_GET();
+    PyFrameObject* f = tstate->frame;
+    while (f) {
+      Py_XINCREF((PyObject*)f->f_code);
```

Comment:
This is going to break 3.11 build.
Please use the correct getters from `python_compat.h` please.